ProcessAPI/ProcessInterface.java

```diff
@@ -0,0 +1,14 @@
+/* Step 1: Identify the user
+ *    Compute Engine Class
+ * 
+ * Step 2: What the user needs to do
+ *      process the input and output sources
+ * Step 3: Build a prototype
+ * 
+ */
+
+public interface ProcessInterface
+{
+    public ProcessResponse processData(Sources input, Sources output);

```

Comment:
This is a completely valid design decision, but it leads to an interesting pattern - the response object will need to know how to accept output data (because this is the only method that takes in an output, it must return something that can handle the actual writing behavior). It's a little bit convoluted, which is why it's unusual - this typically shows up for performance reasons, but this particular computation engine falls pretty squarely into the use case: write() will be called potentially many, many times, and processData is a cross-process but not cross-network boundary (so it's reasonable that the response can actually do the writing) and avoiding cross-process calls in an inner loop is a very reasonable performance-related decision.

I'm not sure if that's what you were intending to do, and you should feel free to split this into two methods (read & write), but if you want to roll with this particular design pattern, that could end being pretty cool (this is a bit of a choose-your-difficulty-level decision for the project).